### PR TITLE
Unbound: now implemented with UCI 

### PR DESCRIPTION
**Pull request by EricLuehrsen**
Maintainer: me
Compile tested: ar71xx OpenWrt 15.05.1
Run tested: TL-WDR3600
Description:
Finally tested and done. See **README.md** 2202548ff0276721017337f6d92ec07a177e5619 as source of requirements. Some things could be improved like procd triggers on interfaces instead of hotplug, but lets save those for another day.

Please withhold pull until @hansmi and @Borromini can scrutinize content; yes the last commit is left divided for now to aid in --fixup.

### CODE REVIEW

File: net/unbound/files/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Unbound Recursive DNS Server with UCI
+
+## Unbound Description
+Unbound is a validating, recursive, and caching DNS resolver. The C implementation of Unbound is developed and maintained by [NLnet Labs](https://www.unbound.net/). It is based on ideas and algorithms taken from a java prototype developed by Verisign labs, Nominet, Kirei and ep.net. Unbound is designed as a set of modular components, so that also DNSSEC (secure DNS) validation and stub-resolvers (that do not run as a server, but are linked into an application) are easily possible.
+
+## Package Overview
+Unbound may be useful on consumer grade embedded hardware. It is *intenteded* to be a recursive resolver only. [NLnet Labs NSD](https://www.nlnetlabs.nl/projects/nsd/) is *intenteded* for the authoritative task. This is different than ISC-Bind and its inclusive functions. As a consequence, Unbound configuration effort and memory consumption may be easier to control. A consumer could have their own recursive resolver, and remove potential issues from forwarding resolvers out side of their control.
```
Comment (hansmi) [inline net/unbound/files/README.md]: `s/intenteded/intended/g`

File: net/unbound/files/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+This package builds on Unbounds capabilities with OpenWrt UCI. Each Unbound option is not in UCI, but rather, UCI is used to simplify the combination of related options. Many of Unbounds native options are balanced and bundled to smaller user choices. Options include resources, DNSEC, access control, and some TTL tweaking. The UCI also provides an option to work at the raw "unbound.conf" level.
```
Comment (hansmi) [inline net/unbound/files/README.md]: > Each Unbound option is not in UCI […]

Replace with “Not every Unbound option is in UCI”.

File: net/unbound/files/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Work with dnsmasq
+The presented options will help Unbound and dnsmasq work together effeciently. Many want internal host identification in DNS. Alternative DHCP event scripts which write host records are difficult to formulate for Unbound, NSD, or Bind. In OpenWrt, dnsmsaq is the default DHCP and DNS stub resolver, and this will continue to serve internal DNS.
+
+Follow these instructions to make Unbound primary for handing DNS, and dnsmasq only serve local network DHCP-DNS. Set "option dnsmasq\_link\_dns" true and other related options as desired within Unbound. It will be necessary then to disable dnsmasq global lookups within "/etc/config/dhcp" and seciton "config dnsmasq." Set "option noresolv" true, remove "option resolvfile," and set "option port" to 1053 or 5353. Also for each "config dhcp \[*name*\], set "list dhcp_option option:dns-server,0.0.0.0." This last defends against internal combinations where dnsmasq may disable DHCP send options when it thinks it may not be performing DNS.
```
Comment (hansmi) [inline net/unbound/files/README.md]: `s/seciton/section/g`

Stopped reading at this point. Please run the document through a spell checker. :-)

File: net/unbound/files/rootzone.sh [hunk 4]
```diff
@@ -21,26 +21,35 @@
...
+  return 0
 }
 
 ##############################################################################
 
 rootkey_update() {
   local basekey_date rootkey_date rootkey_age filestuff
-
-  # TODO: Just structure to real UCI coming soon.
+
```
Comment (hansmi) [inline net/unbound/files/rootzone.sh]: Unecessary EOL whitespace, in other places and files too.

**Comment by Borromini** [comment]
I'm really looking forward to UCI integration for Unbound but between fixing my build setup and preparing for holidays, I haven't found time yet, sorry... And I am leaving tomorrow :( If it's still in review when I get back end of November, I will gladly test.

File: net/unbound/files/rootzone.sh (hunk 4 above)
Comment (EricLuehrsen) [inline net/unbound/files/rootzone.sh]: That seems pretty common for "smart indent" enabled in any editor. Blank lines are left ready at the respective indent level.

**Comment by EricLuehrsen** [comment]
This is significant new content, so I will hold this pull for peer acceptance. In the mean time, I would like any new feedback on your minds, cosmetic or functional.

**Comment by hnyman** [comment]
ok to merge?

**Comment by EricLuehrsen** [comment]
@hnyman, okay you're right, it is lingering, and so go ahead with the merge. I am confident in the work. Note: users can toggle a UCI option to disable itself and use their own conf-file from package 1.5.10-3.

**Pull request merged**